### PR TITLE
Bump dataflow and threading

### PR DESCRIPTION
**Pull request by drewnoakes**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8919)

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb [hunk 1]
```diff
@@ -114,7 +114,7 @@ Namespace Microsoft.VisualStudio.Editors.OptionPages
             ' The main copy of the options has changed. Update the copy used by the
             ' Tools | Options UI, if has been created.
             If _shouldUpdateOptionsControlOnPropertyChange AndAlso _optionsControl IsNot Nothing Then
-                ThreadHelper.JoinableTaskFactory.RunAsync(
+                Dim unused = ThreadHelper.JoinableTaskFactory.RunAsync(
```
Comment (melytc) [inline src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb:117]: why is the variable needed?

**Review by melytc — APPROVED** [review]

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb (hunk 1 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb:117]: This is the VB equivalent of a discard in C# (i.e. `_ = ThreadHelper...`).

In this case `RunAsync` returns a `Task` that is never being awaited. The newer version of the threading analyzers flags this as a warning, as we generally want to await tasks. In cases where we don't wish to await them, we use a discard like this to hint to other developers (and the analyzer) that the task is intentionally being let free without being awaited.

Comment (MiYanni) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb:117]: Likely a warning about the result of it not being assigned. We have `_ = ` in C# for it. I don't think there is an equivalent in VB.

**Review by MiYanni — APPROVED** [review]

File: src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb (hunk 1 above)
Comment (melytc) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb:117]: today I learned 🙂 thanks!

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPage.vb:117]: https://github.com/dotnet/vblang/issues/346

**Pull request merged**